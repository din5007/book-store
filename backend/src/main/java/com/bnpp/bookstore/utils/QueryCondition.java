package com.bnpp.bookstore.utils;

import io.micrometer.common.util.StringUtils;
import org.springframework.stereotype.Component;

@Component("queryUtil")
public class QueryCondition {

  public boolean checkNull(String field) {
    return StringUtils.isBlank(field);
  }
}
