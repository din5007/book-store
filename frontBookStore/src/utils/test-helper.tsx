import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";


function customRender(component : any , store?: any) {

    return render(<StoreProvider store={store}>
        <BrowserRouter>
            {component}
        </BrowserRouter>
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;