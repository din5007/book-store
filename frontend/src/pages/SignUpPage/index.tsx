import { lazy } from 'react';

const SignUpPage = lazy(() => import('./SignUpPage'));

export default SignUpPage;