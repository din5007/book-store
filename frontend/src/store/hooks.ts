import { createTypedHooks } from "easy-peasy";

import { AppStoreModel } from ".";

export const { useStoreActions, useStoreState } = createTypedHooks<AppStoreModel>();