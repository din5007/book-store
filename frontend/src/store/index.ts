import { HttpService } from "../service/http/http.service";
import { createStore } from 'easy-peasy';
import { BookModel, booksModel } from "./books/books.model";
import { cartModel, CartModel } from "./cart/cart.model";
import { userModel, UserModel } from "./users/user.model";
import { OrderModel, orderModel } from "./order/order.model";

export interface AppStoreModel {
    books: BookModel;
    cart: CartModel;
    user: UserModel;
    order: OrderModel;
}

export const appStoreModel : AppStoreModel = {
    books: booksModel,
    cart: cartModel,
    user: userModel,
    order: orderModel,
}

export function storeModel() {
    return createStore(appStoreModel, {
        injections: {
            httpService: new HttpService()
        },
        initialState: {
            
        },
    })
}
export const store =  storeModel();