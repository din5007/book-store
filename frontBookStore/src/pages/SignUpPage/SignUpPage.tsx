
import { useState } from "react";
import Button from "react-bootstrap/Button";
import Col from "react-bootstrap/esm/Col";
import Row from "react-bootstrap/esm/Row";
import Form from "react-bootstrap/Form";
import { Link } from "react-router-dom";
import "./SignUpPage.css";
import ErrorMessage from "../CommonPages/ErrorPage";
import Loading from "../CommonPages/LoadingPage";
import MainPage from "../MainPage/MainPage";
import { useStoreActions } from "../../store/hooks";

const SignUpPage = () => {
  const [email, setEmail] = useState("");
  const [name, setName] = useState("");
  const [password, setPassword] = useState("");
  const [confirmpassword, setConfirmPassword] = useState("");
  const [message, setMessage] = useState("");
  const [error, setError] = useState(false);
  const [loading, setLoading] = useState(false);

  const { signUp } = useStoreActions((actions) => actions.user);

  const submitHandler = async (e : any) => {
    e.preventDefault();

    if (password !== confirmpassword) {
      setMessage("Password Do Not Match");
    } else {
      setMessage("null");
      try {
        setLoading(true);
        signUp({ name, email, password});
        setLoading(false);
      } catch (error : any) {
        setError(error.response.data.message);
      }
    }
  };

  return (
    <MainPage title="Register">
      <div className="loginContainer">
        {error && <ErrorMessage variant="danger">{error}</ErrorMessage>}
        {message && <ErrorMessage variant="danger">{message}</ErrorMessage>}
        {loading && <Loading />}
        <Form onSubmit={submitHandler}>
          <Form.Group className="mb-3" controlId="name">
            <Form.Label>Name</Form.Label>
            <Form.Control
              type="name"
              value={name}
              placeholder="Enter User name"
              onChange={(e) => setName(e.target.value)}
            />
          </Form.Group>

          <Form.Group className="mb-3" controlId="formBasicEmail">
            <Form.Label>Email address</Form.Label>
            <Form.Control
              type="email"
              value={email}
              placeholder="Enter Your Email"
              onChange={(e) => setEmail(e.target.value)}
            />
          </Form.Group>

          <Form.Group className="mb-3" controlId="formBasicPassword">
            <Form.Label>Password</Form.Label>
            <Form.Control
              type="password"
              value={password}
              placeholder="Enter Your Password"
              onChange={(e) => setPassword(e.target.value)}
            />
          </Form.Group>

          <Form.Group className="mb-3" controlId="confirmPassword">
            <Form.Label>Confirm Password</Form.Label>
            <Form.Control
              type="password"
              value={confirmpassword}
              placeholder="Enter Your Confirm Password"
              onChange={(e) => setConfirmPassword(e.target.value)}
            />
          </Form.Group>
          <Button variant="primary" type="submit">
            Register
          </Button>
        </Form>
        <Row className="py-3">
          <Col>
            Have an Account Already ? <Link to="/login">Login</Link>
          </Col>
        </Row>
      </div>
    </MainPage>
  );
};

export default SignUpPage;
