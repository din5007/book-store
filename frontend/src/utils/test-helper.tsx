import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { AppStoreModel, store } from "../store";
import { Router } from 'react-router-dom';
import { ReactNode } from "react";



// const provider : React.FC<{children : ReactNode, store: any}> = ({ children, store }) => {
//     return (
        
//     )
// }

function customRender(component : any , store: any) {

    return render(<StoreProvider store={store}>
        {component}
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;