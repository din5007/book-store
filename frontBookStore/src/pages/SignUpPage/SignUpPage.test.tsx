import { createStore } from 'easy-peasy';
import { customRender as render } from '../../utils/test-helper';
import SignUpPage from "./SignUpPage"
import { describe, expect, it, vi } from 'vitest';
import { appStoreModel } from '../../store';

const mock = {
    get: vi.fn()
};

const store = createStore(appStoreModel, {
    injections: {
        httpService : mock
    },
    initialState: {
        
    }
})

describe('SignUp Page', () => {
    it('SignUp Page should load properly - snapshot', async () => {
        const {container}  = render(<SignUpPage />, store);
        expect(container).toMatchSnapshot();
    });
})