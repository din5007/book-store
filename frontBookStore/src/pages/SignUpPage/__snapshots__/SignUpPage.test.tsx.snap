// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignUp Page > SignUp Page should load properly - snapshot 1`] = `
<div>
  <div
    style="min-height: 93vh; display: flex; padding: 10px 0px;"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="w-100"
        >
          <h1>
            Register
          </h1>
          <div
            style="display: flex; flex-direction: column; margin: 20px;"
          >
            <form
              class=""
            >
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="name"
                >
                  Name
                </label>
                <input
                  class="form-control"
                  id="name"
                  placeholder="Enter User name"
                  type="name"
                  value=""
                />
              </div>
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="formBasicEmail"
                >
                  Email address
                </label>
                <input
                  class="form-control"
                  id="formBasicEmail"
                  placeholder="Enter Your Email"
                  type="email"
                  value=""
                />
              </div>
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="formBasicPassword"
                >
                  Password
                </label>
                <input
                  class="form-control"
                  id="formBasicPassword"
                  placeholder="Enter Your Password"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="confirmPassword"
                >
                  Confirm Password
                </label>
                <input
                  class="form-control"
                  id="confirmPassword"
                  placeholder="Enter Your Confirm Password"
                  type="password"
                  value=""
                />
              </div>
              <button
                class="btn btn-primary"
                type="submit"
              >
                Register
              </button>
            </form>
            <div
              class="py-3 row"
            >
              <div
                class="col"
              >
                Have an Account Already ? 
                <a
                  href="/login"
                >
                  Login
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
